src/client.rs

```diff
@@ -58,23 +62,65 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         }))
         .await?;
 
-    let input_config = netavark_proxy::g_rpc::NetworkConfig::load(&file)?;
-
     let client = NetavarkProxyClient::new(channel);
 
     let result = match opts.subcmd {
         SubCommand::Setup(_) => {
             let s = setup::Setup::new(input_config);
-            s.exec(client).await?
+            s.exec(client).await
         }
         SubCommand::Teardown(_) => {
             let t = teardown::Teardown::new(input_config);
-            t.exec(client).await?
+            t.exec(client).await
+        }
+    };
+
+    let r = match result {
+        Ok(r) => r,
+        Err(e) => {
+            eprintln!("Error: {}", e.message());
+            process_failure(e)
         }
     };
-    let pp = ::serde_json::to_string_pretty(&result);
+
+    let pp = ::serde_json::to_string_pretty(&r.into_inner());
     // TODO this should probably return an empty lease so consumers
     // don't soil themselves
     println!("{}", pp.unwrap_or_else(|_| "".to_string()));
     Ok(())
 }
+
+//
+// process_failure makes the client exit with a specific
+// error code
+//
+fn process_failure(status: Status) -> tonic::Response<Lease> {
+    let mut rc: i32 = 1;
+
+    match status.code() {
+        Code::Ok => {}
+        Code::Cancelled => {}
+        Code::Unknown => {
+            rc = 155;
+        }
+        Code::InvalidArgument => {
+            rc = 156;
+        }
+        Code::DeadlineExceeded => {}
+        Code::NotFound => {
+            rc = 6;
+        }
+        Code::AlreadyExists => {}
```

Comment:
if you do not plan to implement most of them it would be cleaner to just `_ => {}` which acts as default case
so you could only define the ones you need

Reply:
fixed in next pr